packages/notte-sdk/src/notte_sdk/websockets/base.py

```
@@ -0,0 +1,117 @@
+
+    def _run_async_loop(self) -> None:
+        """Run the async event loop in a separate thread."""
+        self._loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self._loop)
+        try:
+            # Create a task that we can cancel
+            self._ws_task = self._loop.create_task(self.watch())
+            self._loop.run_until_complete(self._ws_task)  # pyright: ignore [reportUnknownMemberType, reportUnknownArgumentType]
+        except asyncio.CancelledError:
+            pass  # Task was cancelled, which is expected during shutdown
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)